staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -31,32 +31,37 @@ import (
 	"k8s.io/apiserver/pkg/storage/value"
 	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/kms"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/secretbox"
 )
 
 const (
 	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
 	aesGCMTransformerPrefixV1    = "k8s:enc:aesgcm:v1:"
 	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1:"
+	kmsTransformerPrefixV1       = "k8s:enc:kms:v1:"
 )
 
-// GetTransformerOverrides returns the transformer overrides by reading and parsing the encryption provider configuration file
-func GetTransformerOverrides(filepath string) (map[schema.GroupResource]value.Transformer, error) {
+type kmsServiceGetter func(name string, kmsConfig map[string]interface{}) (kms.Service, error)
+
+// GetTransformerOverrides returns the transformer overrides by reading and parsing the encryption provider configuration file.
+// It takes an optional kmsServiceGetter as an argument, which is used when a KMS based encryption backend is used.
+func GetTransformerOverrides(filepath string, initKMSService kmsServiceGetter) (map[schema.GroupResource]value.Transformer, error) {
 	f, err := os.Open(filepath)
 	if err != nil {
 		return nil, fmt.Errorf("error opening encryption provider configuration file %q: %v", filepath, err)
 	}
 	defer f.Close()
 
-	result, err := ParseEncryptionConfiguration(f)
+	result, err := ParseEncryptionConfiguration(f, initKMSService)
```

Comment:
I did not understand "bind the config prior to passing it here". Since the `initKMSService` function is responsible for accessing the registered KMS providers, we cannot know what all plugins are available, without talking to this function. What we can do, is to add a new member `Init func(config map[string]interface{}) error` to `kms.Service` interface. That way, `kmsServiceGetter` would become `kmsServiceGetter func(name string) kms.Service`.

The code [here](https://github.com/sakshamsharma/kubernetes/blob/e8613c6f0b61236399c33917b8f82b0b8bd37284/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L152) would then look like:
```
kmsService, err := initKMSService(kind).Init(provider.KMS.Config)
```

The reason we cannot do it the way it was done for cloud providers, is that we do not have the KMS configuration available before calling `GetTransformerOverrides`.

Can you please clarify and let me know, if I mis-interpreted your comment?